src/IO/AsciiPCLoader/asciipcloader.cpp

```diff
@@ -0,0 +1,203 @@
+using std::to_string;
+
+const string trajExt( "txt" );
+
+namespace Ra {
+namespace IO {
+
+AsciiPointCloudLoader::AsciiPointCloudLoader() {}
+
+AsciiPointCloudLoader::~AsciiPointCloudLoader() {}
+
+vector<string> AsciiPointCloudLoader::getFileExtensions() const {
+    return vector<string>( { "*." + trajExt } );
+}
+
+bool AsciiPointCloudLoader::handleFileExtension( const string& extension ) const {
+    return extension.compare( trajExt ) == 0;
+}
+
+Ra::Core::Asset::FileData* AsciiPointCloudLoader::loadFile( const string& filename ) {
+    using Ra::Core::Asset::FileData;
+    using Ra::Core::Asset::GeometryData;
+    using Ra::Core::Utils::logERROR;
+    using Ra::Core::Utils::logINFO;
+
+    FileData* fileData = new Ra::Core::Asset::FileData( filename );
+
+    if ( !fileData->isInitialized() ) {
+        delete fileData;
+        LOG( logINFO ) << "Filedata could not be initialized";
+        return nullptr;
+    }
+
+    auto startTime = clock();
+    // a unique name is required by the component messaging system
+    static int id = 0;
+    auto geometry =
+        make_unique<GeometryData>( "TRAJ_PC_" + to_string( ++id ), GeometryData::POINT_CLOUD );
+    if ( geometry == nullptr ) {
+        LOG( logERROR ) << "could not create geometry";
+        return nullptr;
+    }
+
+    geometry->setFrame( Ra::Core::Transform::Identity() );
+
+    ifstream stream( filename );
+    if ( !stream.is_open() ) {
+        throw runtime_error( "Could not open ifstream to path " + filename );
+        return nullptr;
+    }
+    else {
+        LOG( logINFO ) << "---Beginning file loading---";
+    }
+
+    /*reading how data is arranged*/
+    string line;
+    vector<unsigned long> spacepos;
+    unsigned long timepos, xpos, ypos, zpos;
+
+    if ( !getline( stream, line ) ) {
+        delete fileData;
+        LOG( logINFO ) << "file does not contain trajectory data format. aborting";
+        return nullptr;
+    }
+
+    timepos = line.find( "Time" );
+    xpos    = line.find( " X" );
+    ypos    = line.find( " Y" );
+    zpos    = line.find( " Z" );
+
+    if ( timepos == string::npos ) {
+        delete fileData;
+        LOG( logINFO ) << "file does not contain time property. aborting";
+        return nullptr;
+    }
+
+    if ( xpos == string::npos ) {
+        delete fileData;
+        LOG( logINFO ) << "file does not contain X property. aborting";
+        return nullptr;
+    }
+
+    if ( ypos == string::npos ) {
+        delete fileData;
+        LOG( logINFO ) << "file does not contain Y property. aborting";
+        return nullptr;
+    }
+
+    if ( zpos == string::npos ) {
+        delete fileData;
+        LOG( logINFO ) << "file does not contain Z property. aborting";
+        return nullptr;
+    }
+
+    LOG( logINFO ) << "Loading properties \"x\", \"y\", \"z\", \"time\".";
+
+    // skipping blank space
+    xpos++;
+    ypos++;
+    zpos++;
+
+    /*retrieving x y z time attribut positions and column count*/
```

Comment:
With this implementation, your loader can only load the x,y,z and time fields.
I would be better to load all the fields, then aggregate x,y,z, and add the position, and custom fields in the Radium datastructure.

Reply:
updated implementation to load all attributes as custom and aggregate x y z properties only